easybuild/framework/easyconfig/default.py

```diff
@@ -107,18 +109,23 @@
     'github_account': ['%(namelower)s', "GitHub account name to be used to resolve template values in source URLs",
                        BUILD],
     'hidden': [False, "Install module file as 'hidden' by prefixing its version with '.'", BUILD],
-    'installopts': ['', 'Extra options for installation', BUILD],
+    'installopts': ['', 'Extra options appended to installation command', BUILD],
+    'install_cmd': [None, "Main shell command used in the install step", BUILD],
     'maxparallel': [None, 'Max degree of parallelism', BUILD],
     'parallel': [None, ('Degree of parallelism for e.g. make (default: based on the number of '
                         'cores, active cpuset and restrictions in ulimit)'), BUILD],
     'patches': [[], "List of patches to apply", BUILD],
-    'prebuildopts': ['', 'Extra options pre-passed to build command.', BUILD],
-    'preconfigopts': ['', 'Extra options pre-passed to configure.', BUILD],
-    'preinstallopts': ['', 'Extra prefix options for installation.', BUILD],
-    'pretestopts': ['', 'Extra prefix options for test.', BUILD],
-    'postinstallcmds': [[], 'Commands to run after the install step.', BUILD],
-    'postinstallpatches': [[], 'Patch files to apply after running the install step.', BUILD],
-    'postinstallmsgs': [[], 'Messages to print after running the install step.', BUILD],
+    'pre_build_cmds': ['', 'Extra commands executed before main build command', BUILD],
```

Comment:
Mention that these are run in a different subshell than the build command itself (no sharing of environment), and only when there's no generic/custom easyblock is used

Same for `configure`, `test`, `install`